pkg/reconciler/buildrun/resources/conditions.go

```diff
@@ -97,7 +97,10 @@ func UpdateBuildRunUsingTaskRunCondition(ctx context.Context, client client.Clie
 			//nolint:staticcheck // SA1019 we want to give users some time to adopt to failureDetails
 			buildRun.Status.FailedAt = &buildv1alpha1.FailedAt{Pod: pod.Name}
 
-			if failedContainer != nil {
+			if pod.Status.Reason == "Evicted" {
+				message = pod.Status.Message
+				reason = buildv1alpha1.BuildRunStatePodEvicted
+			} else if failedContainer != nil {
```

Comment:
Can it be that the Pod is evicted and we would still have a failedContainer (= the one that caused the eviction though too much ephemeral storage usage) ? If so, we would want to still capture it in `buildRun.Status.FailedAt.Container` ?

Reply:
So in one of our usecases we had exactly the same buildrun.
It was failing twice. In both cases it was evicted due to ephemeral storage.
In one case shipwright was detecting a failed Container. In the other it wasn't 
But the failed container didn't had any helpful information. 

Unfortunately I don't have the yamls of those resources anymore

Reply:
>  If so, we would want to still capture it in buildRun.Status.FailedAt.Container ?
Based on the information I just provided I'm not sure if we really need this information or not 

In the end it's enough to know that the pod got evicted due to resource constraints. The user would then need to update their resource requests and rerun

Reply:
fine with that. probably doesn't hurt to have more information